### DISCUSSION TITLE
Using Revise and LoweredCodeUtils

### DISCUSSION THREAD

**Discussion by karlwessel** [fonsp/Pluto.jl#2069] [Pluto development]
Would it make sense to use Revise for updating methods and [LoweredCodeUtils](https://juliadebug.github.io/LoweredCodeUtils.jl/stable/edges/) for calculating cell dependencies instead of implementing it yourself?

This way Pluto could benefit from the stability of Revise for parsing and evaluating Julia code.

**Comment by Pangoraw** [comment]
Thanks for sharing! One problem with Revise is that it would still require some work to integrate with the way Pluto currently allows to redefine structs.

**Comment by karlwessel** [comment]
Is the current solution to redefine structs not compatible with revise?

To get an idea on how much work this would be: can you give me some pointers which parts of Pluto I would have to touch, replace or interface with if I would like to play around with this idea?

**Reply by Pangoraw** [reply]
A good place to start is [`run_reactive!`](https://github.com/fonsp/Pluto.jl/blob/f03f73ca2e345dfb3138521bcf48ac8c06250390/src/evaluation/Run.jl#L39) which runs all the cells and its downstream cells while creating a new Julia module to hold the new structs.
